sycl/test/extensions/atomic16/atomic_half.cpp

```diff
@@ -0,0 +1,34 @@
+// RUN: %clangxx %s -fsycl-device-only -S -o - | FileCheck %s
+
+// CHECK: call spir_func void @_Z19__spirv_AtomicStore{{.*}}(ptr {{.*}}, i32 {{.*}}, i32 {{.*}}, half noundef 0xH3C00)
+// CHECK: call spir_func noundef half @_Z18__spirv_AtomicLoad{{.*}}(ptr {{.*}}, i32 {{.*}}, i32 {{.*}})
+// CHECK: call spir_func noundef half @_Z22__spirv_AtomicExchange{{.*}}(ptr {{.*}}, i32 {{.*}}, i32 {{.*}}, half {{.*}})
+// CHECK: call spir_func noundef half @_Z21__spirv_AtomicFAddEXT{{.*}}(ptr {{.*}}, i32 {{.*}}, i32 {{.*}}, half {{.*}})
+// CHECK: call spir_func noundef half @_Z21__spirv_AtomicFAddEXT{{.*}}(ptr {{.*}}, i32 {{.*}}, i32 {{.*}}, half {{.*}})
+// CHECK: call spir_func noundef half @_Z21__spirv_AtomicFMinEXT{{.*}}(ptr {{.*}}, i32 {{.*}}, i32 {{.*}}, half {{.*}})
+// CHECK: call spir_func noundef half @_Z21__spirv_AtomicFMaxEXT{{.*}}(ptr {{.*}}, i32 {{.*}}, i32 {{.*}}, half {{.*}})
+
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue q;
+  q.submit([&](sycl::handler &h) {
+    h.single_task([=]() {
```

Comment:
I remember a series of patches intended to simplify our tests. Strictly speaking, you do not need a kernel here, you can get away with simple `SYCL_EXTERNAL` function. By doing so you should be able to replace `#include <sycl/sycl.hpp>` with a couple of more precise includes, thus potentially speeding up the test

Reply:
We did that for `check_device_code` tests because `CHECK`s are much simpler with `SYCL_EXTERNAL` approach (if we're referring to the same set of changes). As for `sycl/sycl.hpp` we only have the restriction on its usage for E2E tests (and it's being enforced by a dedicated test there).

I'm not saying that either of the two can't be applied under `sycl/test` "voluntarily" though :)

Reply:
I guess at least I had to use sycl/core.hpp here